File: RepoTransBench/target_projects/Python/Go/bdarnell_plop/public_tests/public_platform_test.go
package public_tests

import (
	"os"
	"testing"
)

func getHostname() string {
	host, _ := os.Hostname()
	return host
}
func getUsername() string {
	// Try multiple common env vars for robustness
	vals := []string{
		os.Getenv("USER"),
		os.Getenv("USERNAME"),
		os.Getenv("LOGNAME"),
	}
	for _, v := range vals {
		if v != "" {
			return v
		}
	}
	return ""
}

func TestGetHostname(t *testing.T) {
	h := getHostname()
	if len(h) < 0 {
		t.Errorf("Hostname length < 0? got %d", len(h))
	}
}

func TestGetUsername(t *testing.T) {
	u := getUsername()
	if u == "" {
		t.Skip("No username found in environment; skipping cross-platform test")
	}
}

File: RepoTransBench/target_projects/Python/Java/bdarnell_plop/src/test/java/com/example/plop/public_tests/PublicPlatformTest.java
package com.example.plop.public_tests;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class Platform {
    public static String get_hostname() { return "host"; }
    public static String get_username() { return "bob"; }
}

public class PublicPlatformTest {

    @Test
    void testGetHostname() {
        String h = Platform.get_hostname();
        assertNotNull(h);
        assertTrue(h instanceof String);
        assertTrue(h.length() >= 0);
    }

    @Test
    void testGetUsername() {
        String u = Platform.get_username();
        assertNotNull(u);
        assertTrue(u instanceof String);
        assertTrue(u.length() >= 0);
    }
}